File: problem1.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        if (nums == null && nums.length == 0) {
            return 0;
        }
        int n = nums.length;
        int min = Integer.MAX_VALUE;
        PriorityQueue<Integer> pq = new PriorityQueue<>((a,b) -> b - a);
        for (int i = 0; i < n; i++) {
            int num = nums[i];
            pq.add(num);
            if (pq.size() > n-k) {
                min = Math.min(min, pq.poll());
            }
        }
        return min;
    }
}

File: problem1.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        if nums is None and len(nums) == 0:
            return 0
        n = len(nums)
        pq = []
        for i in range(n):
            num = nums[i]
            heapq.heappush(pq, num)
            if len(pq) > k:
                heapq.heappop(pq)
        return heapq.heappop(pq)